api/app/GraphQL/Validators/CreateUpdateWorkExperienceValidator.php

```diff
@@ -18,9 +19,9 @@ final class CreateUpdateWorkExperienceValidator extends Validator
     public function rules(): array
     {
         return [
-            // 'workExperience.employmentCategory' => [
-            //     'required',
-            // ],
+            'workExperience.employmentCategory' => [
+                'required',
+            ],
```

Comment:
Linking skills to an experience also uses this mutation, so the validator then trips up the Skills requirement step of an application. In that step the experience was already created, so you are editing it effectively.
Not sure what the approach should be here 🫤 

Also the success toast is misleading, it failed but that is a separate problem  

![image](https://github.com/user-attachments/assets/83578fd0-471a-41f8-b99b-e8393b2a6d15)

Reply:
The toast stems from below, the issue is `data` is truthy, just a null value connected to a key

![image](https://github.com/user-attachments/assets/275ad04f-371f-442f-af14-698ba30ca3bc)

Elsewhere we handle this by checking that `error` is null or querying into `data`

![image](https://github.com/user-attachments/assets/e8ca9cb4-6fa1-4b6b-9940-bc44c92d5f4f)

![image](https://github.com/user-attachments/assets/45109722-7897-4760-a20e-110c2ba5f9c2)

Reply:
Nice catch :cry: I'm not sure where to from here. I think we can change the validation to use `sometimes`, so it [only validates it when present](https://laravel.com/docs/11.x/validation#validating-when-present). I'll check with @tristan-orourke.

Reply:
I tried it out in [276b5e2](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12208/commits/276b5e210eeea535b18a9c8fadd29d01ebfee639) and it seems to be working.

Reply:
I just chatted with Tristan. He said using the `sometimes` rule is fine for now. However, like Peter mentioned, we should make a ticket to create a separate mutation.

Rule added in [276b5e2](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12208/commits/276b5e210eeea535b18a9c8fadd29d01ebfee639)